src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -533,16 +533,16 @@ protected BoundExpression BindExpression(ExpressionSyntax node, BindingDiagnosti
         protected BoundExpression BindExpressionAllowArgList(ExpressionSyntax node, BindingDiagnosticBag diagnostics)
         {
             BoundExpression expr = BindExpressionInternal(node, diagnostics, invoked: false, indexed: false);
-            VerifyUnchecked(node, diagnostics, expr);
+            CheckContextForPointerTypes(node, diagnostics, expr);
             return expr;
         }
 
-        private void VerifyUnchecked(ExpressionSyntax node, BindingDiagnosticBag diagnostics, BoundExpression expr)
+        private void CheckContextForPointerTypes(ExpressionSyntax node, BindingDiagnosticBag diagnostics, BoundExpression expr)
         {
             if (!expr.HasAnyErrors && !IsInsideNameof)
             {
                 TypeSymbol exprType = expr.Type;
-                if ((object)exprType != null && exprType.IsUnsafe())
+                if ((object)exprType != null && exprType.ContainsPointer())
```

Comment:
Nit.  "IsOrContainsPointerType" feels like a clearer name to me.  But i can accept ContainsPointer as being clear enough as well.

Reply:
I'll keep `ContainsPointer` because that name aligns with other methods we have, like `ContainsDynamic`, `ContainsNativeIntegerWrapperType`, `ContainsErrorType`, etc